src/app/components/SelectWithSearch.tsx

```diff
@@ -34,6 +38,9 @@ export default function SelectWithSearch(props: SelectWithSearchProps) {
         setSelected(value)
         setSearch("")
         props.onChange(value)
+
+        const toStorageValue = { name: value.name, code: value.code }
```

Comment:
toStorageValue looks like a function name, why don't you just call it `storageValue` ?

Reply:
Removed from this file to keep SelectWithSearch as generic as possible.